fil-proofs-tooling/src/bin/benchy.rs

```diff
@@ -0,0 +1,636 @@
+                        .long("no-tmp")
+                        .help("Don't use a temp file for random data (write to current directory instead).")
+                )
+                .arg(
+                    Arg::with_name("dump")
+                        .long("dump")
+                        .help("Dump vanilla proofs to current directory.")
+                )
+                .arg(
+                    Arg::with_name("partitions")
+                        .long("partitions")
+                        .help("How many circuit partitions to use")
+                        .default_value("1")
+                        .takes_value(true),
+                )
+                .arg(
+                    Arg::with_name("groth")
+                        .long("groth")
+                        .help("Generate and verify a groth circuit proof.")
+                )
+                .arg(
+                    Arg::with_name("bench")
+                        .long("bench")
+                        .help("Synthesize and report inputs/constraints for a circuit.")
+                )
+                .arg(
+                    Arg::with_name("no-bench")
+                        .long("no-bench")
+                        .help("Don't synthesize and report inputs/constraints for a circuit.")
+                )
+                .arg(
+                    Arg::with_name("bench-only")
+                        .long("bench-only")
+                        .help("Don't replicate or perform Groth proving.")
+                        .conflicts_with_all(&["no-bench", "groth", "extract"])
+                )
+                .arg(
+                    Arg::with_name("circuit")
+                        .long("circuit")
+                        .help("Print the constraint system.")
+                )
+                .arg(
+                    Arg::with_name("extract")
+                        .long("extract")
+                        .help("Extract data after proving and verifying.")
+                )
+                .arg(
+                    Arg::with_name("taper")
+                        .long("taper")
+                        .help("fraction of challenges by which to taper at each layer")
+                        .default_value("0.0")
+                )
+                .arg(
+                    Arg::with_name("taper-layers")
+                        .long("taper-layers")
+                        .help("number of layers to taper")
+                        .takes_value(true)
+                )
+                .arg(
+                    Arg::with_name("push-prometheus")
+                        .long("push-prometheus")
+                        .help("When enabled pushes the collected stats to a prometheus pushgateway")
+                )
+        )
+        .get_matches();
+
+    match matches.subcommand() {
+        ("zigzag", Some(m)) => zigzag_cmd(m),
+        _ => panic!("carnation"),
+    }
+    .expect("command failed")
+}
+
+lazy_static! {
```

Comment:
please put the zigzag details into its own file, so I can add others easily

Reply:
done!